client/src/main/java/io/pravega/client/byteStream/impl/ByteStreamClientImpl.java

```diff
@@ -48,7 +51,11 @@
 
     @Override
     public ByteStreamReader createByteStreamReader(String streamName) {
-        return createByteStreamReaders(new Segment(scope, streamName, 0));
+        // Fetch the segments pointing to the current HEAD of the stream.
+        Map<Segment, Long> segments = Futures.getThrowingException(controller.getSegmentsAtTime(Stream.of(scope, streamName), 0L));
```

Comment:
Will this correctly handle truncated streams? Is timestamp here the same as epoch?

Reply:
yes, this API returns the segment pointing to the HEAD of the stream even for truncated streams.